aws/table_aws_cloudtrail_trail.go

```diff
@@ -387,6 +387,48 @@ func getCloudtrailTrailEventSelector(ctx context.Context, d *plugin.QueryData, h
 	return item, nil
 }
 
+func getCloudtrailTrailInsightSelector(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	getCommonColumnsCached := plugin.HydrateFunc(getCommonColumns).WithCache()
+	commonData, err := getCommonColumnsCached(ctx, d, h)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_cloudtrail_trail.getCloudtrailTrailInsightSelector", "common_data_error", err)
+		return nil, err
+	}
+	commonColumnData := commonData.(*awsCommonColumnData)
+	trail := h.Item.(types.Trail)
+
+	// Avoid api call if accountId is not equal to the accountId available in arn
+	accountId := arnToAccountId(*trail.TrailARN)
+	if commonColumnData.AccountId != accountId {
+		return nil, nil
+	}
+
+	// Create session
+	svc, err := CloudTrailRegionsClient(ctx, d, *trail.HomeRegion)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_cloudtrail_trail.getCloudtrailTrailInsightSelector", "client_error", err)
+		return nil, err
+	}
+
+	params := &cloudtrail.GetInsightSelectorsInput{
+		TrailName: trail.TrailARN,
+	}
+
+	// List resource tags
+	item, err := svc.GetInsightSelectors(ctx, params)
+	if err != nil {
+		var ae smithy.APIError
+		if errors.As(err, &ae) {
+			if helpers.StringSliceContains([]string{"TrailNotFoundException", "CloudTrailARNInvalidException", "InsightNotEnabledException"}, ae.ErrorCode()) {
```

Comment:
Do we need the `TrailNotFoundException` and `CloudTrailARNInvalidException` errors in this list?

I am assuming it is automatically handled in the List and Get configurations.

Reply:
- In this place we will not get these typo exceptions because we are not passing the input by hand. But the API throws such error if we will pass an invalid input.
- I don't think it will handled in list/get configuration as it is a separate API call, those configurations are particular to list/get API call